pcdsdevices/gon.py

```diff
@@ -236,23 +255,348 @@ class Kappa(BaseInterface, Device):
+                                       180 / 2.0) / np.sin(kappa_ang))
+        k_phi = e_phi * np.pi / 180 - delta
+
+        k_eta = k_eta * 180 / np.pi
+        k_kap = k_kap * 180 / np.pi
+        k_phi = -k_phi * 180 / np.pi
+        return k_eta, k_kap, k_phi
+
+    def mv_e_eta(self, value):
+        """
+        Change the e_eta position to value, keeping e_chi and e_phi the same.
+
+        Parameters
+        ----------
+        value : number
+            Position value to set e_eta to.
+        """
+        try:
+            k_eta, k_kap, k_phi = self.e_to_k(e_eta=value)
+            if self.check_motor_step(k_eta, k_kap, k_phi):
+                logger.info("Starting now moving things!!!")
+                self.eta.mv(k_eta)
+                self.kappa.mv(k_kap)
+                self.phi.mv(k_phi)
+                self.wait()
+        except KeyboardInterrupt:
+            logger.info("Motion interrupted by ctrl+c")
+            self.stop()
+
+    def mv_e_chi(self, value):
+        """
+        Change the e_chi position to value, keeping e_eta and e_phi the same.
+
+        Parameters
+        ----------
+        value : number
+            Position value to set e_chi to.
+        """
+        try:
+            if value == 0:
+                value = 10e-9
+            k_eta, k_kap, k_phi = self.e_to_k(e_chi=value)
+            if self.check_motor_step(k_eta, k_kap, k_phi):
+                logger.info("Starting now moving things!!!")
+                self.eta.mv(k_eta)
+                self.kappa.mv(k_kap)
+                self.phi.mv(k_phi)
+                self.wait()
+        except KeyboardInterrupt:
+            logger.info("Motion interrupted by ctrl+c")
+            self.stop()
+
+    def mv_e_phi(self, value):
+        """
+        Change the e_phi position to value, keeping e_eta and e_phi the same.
+
+        Parameters
+        ----------
+        value : number
+            Position value to set the e_phi to.
+        """
+        try:
+            k_eta, k_kap, k_phi = self.e_to_k(e_phi=value)
+            if self.check_motor_step(k_eta, k_kap, k_phi):
+                logger.info("Starting now moving things!!!")
+                self.eta.mv(k_eta)
+                self.kappa.mv(k_kap)
+                self.phi.mv(k_phi)
+        except KeyboardInterrupt:
+            logger.info("Motion interrupted by ctrl+c")
+            self.stop()
+
+    @pseudo_position_argument
+    def forward(self, pseudo_pos):
+        """
+        Calculate a RealPosition from a given PseudoPosition.
+
+        Parameters
+        ----------
+        pseudo_pos : PseudoPosition
+            The pseudo position input.
+        Returns
+        -------
+        real_position : RealPosition
+            The real position output.
+        """
+        pseudo_pos = self.PseudoPosition(*pseudo_pos)
+        eta, kappa, phi = self.e_to_k(pseudo_pos.e_eta, pseudo_pos.e_chi,
+                                      pseudo_pos.e_phi)
```

Comment:
Should I have here something like this:

```
if self.check_motor_step(eta, kappa, phi):
    return self.RealPosition(......
```
for now only the `mv_e_chi`, `mv_e_eta` and `mv_e_phi` are checking for the step...

Reply:
have added the `check_motor_step` in the `move` method